### DISCUSSION TITLE
Unable to check out process from the pool error

### DISCUSSION THREAD

**Discussion by ytakaya** [supabase/supabase#23707] [Questions]
Hi there,

I’m building a web app using Next.js on vercel, prisma and supabase.
Sometimes the following errors occur in pooler log.
```
ClientHandler: Unable to check out process from the pool due to timeout
```

I thought the pool size was not enough, so I changed it from 15 to 30, but this has not solved the problem.
Should I increase it more?

In addition, when I run load tests, the following errors have been found when requests increase.
(This error is mentioned in [this discussion](https://github.com/orgs/supabase/discussions/18986))
```
ClientHandler: Max client connections reached
```
I am wondering why the errors are different, are these errors caused by different reasons?

**Comment by GaryAustin1** [comment]
A Supabase tips discussion.
https://github.com/orgs/supabase/discussions/22305

**Reply by ytakaya** [reply]
Thanks for information.
I learnt that this log is output when using session mode.
```
ClientHandler: Max client connections reached
```

But I am using transaction mode (port 6543) and it is strange that this log is output...

**Reply by GaryAustin1** [reply]
Not my thing, but I believe you can still overwhelm the supavisor client side connection limit in transacation mode if you have too many connections.  It just makes it so it takes more than the direct connection to the database to do it.

![image](https://github.com/supabase/supabase/assets/54564956/4edf587d-0111-4319-8bef-3999f43dc72a)

**Reply by ytakaya** [reply]
I see.
Last question. Do you know in which case [this log](https://github.com/supabase/supavisor/blob/f468217905db6685ebbc9d3f742b46ef35ce2062/lib/supavisor/client_handler.ex#L607) is output?
I can't reproduce this.
```
ClientHandler: Unable to check out process from the pool due to timeout
```

**Reply by camsloanftc** [reply]
> ClientHandler: Unable to check out process from the pool due to timeout

@ytakaya I am also experiencing this right now on my project - did you ever find a fix?

**Reply by ytakaya** [reply]
@camsloanftc 
Sorry for late.
Once I increased the Pool Size, the error stopped, but the root cause is not known yet.
```
Project Settings > Database > Connection pooling configuration > Pool Size
```

**Reply by Logannford** [reply]
+1 for this as it fixed my issue! Thanks
